org.jacoco.core/src/org/jacoco/core/internal/instr/InstrSupport.java

```diff
@@ -158,18 +158,50 @@ private InstrSupport() {
 	 */
 	static final int CLINIT_ACC = Opcodes.ACC_SYNTHETIC | Opcodes.ACC_STATIC;
 
-	private static final int MAJOR_VERSION_INDEX = 6;
+	/**
+	 * Gets major version number from given bytes of class (unsigned two bytes
+	 * at offset 6).
+	 *
+	 * @return major version of bytecode
+	 * @see <a href=
+	 *      "https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-4.html#jvms-4.1">Java
+	 *      Virtual Machine Specification §4 The class File Format</a>
+	 * @see #setMajorVersion(int, byte[])
+	 * @see #getMajorVersion(ClassReader)
+	 */
+	static int getMajorVersion(final byte[] b) {
+		return ((b[6] & 0xFF) << 8) | (b[7] & 0xFF);
+	}
 
 	/**
-	 * Gets major of bytecode version number from given bytes of class.
+	 * Sets major version number in given bytes of class (unsigned two bytes at
+	 * offset 6).
 	 *
+	 * @param majorVersion
+	 *            major version of bytecode to set
 	 * @param b
 	 *            bytes of class
-	 * @return version of bytecode
+	 * @see #getMajorVersion(byte[])
+	 */
+	static void setMajorVersion(final int majorVersion, final byte[] b) {
+		b[6] = (byte) (majorVersion >>> 8);
+		b[7] = (byte) majorVersion;
+	}
+
+	/**
+	 * Gets major of bytecode version number from given {@link ClassReader}.
+	 *
+	 * @param reader
+	 *            reader to get information about the class
+	 * @return major version of bytecode
+	 * @see ClassReader#ClassReader(byte[], int, int)
+	 * @see #getMajorVersion(byte[])
```

Comment:
This give me a warning because public API links to package private API (which is not part of JavaDoc). Remove link or make methods public?

Reply:
Done

Reply:
@Godin Now as it is public we have a new warning as parameter `b` is not documented 🙈

Reply:
Fixed